client/allocrunner/taskrunner/template/template.go

```diff
@@ -215,6 +216,60 @@ func (tm *TaskTemplateManager) run() {
 	}
 	tm.config.EnvBuilder.SetTemplateEnv(envMap)
 
+	if restored && events != nil {
+		restart := false
+		signals := make(map[string]struct{})
+		for id, event := range events {
+			if !event.DidRender {
+				continue
+			}
+
+			// Lookup the template and determine what to do
+			tmpls, ok := tm.lookup[id]
+			if ok {
+				for _, tmpl := range tmpls {
+					switch tmpl.ChangeMode {
+					case structs.TemplateChangeModeSignal:
+						signals[tmpl.ChangeSignal] = struct{}{}
+					case structs.TemplateChangeModeRestart:
+						restart = true
+					case structs.TemplateChangeModeNoop:
+						continue
+					}
+				}
+			}
+		}
+
+		if restart || len(signals) != 0 {
```

Comment:
We can remove this outer if I think

Reply:
yes here you right, this condition is unnecessary